react/src/views/AutomaticBaseSwitcher/AutomaticBaseSwitcher.tsx

```diff
@@ -2,35 +2,43 @@ import { useAuth0 } from "@auth0/auth0-react";
 import { Button } from "@chakra-ui/react";
 import { GlobalPreferencesContext } from "providers/GlobalPreferencesProvider";
 import React, { useContext, useEffect, useState } from "react";
-import { useNavigate } from "react-router-dom";
+import { useLocation, useNavigate } from "react-router-dom";
 
-const AutomaticBaseSwitcher = () => {
+function AutomaticBaseSwitcher() {
   const navigate = useNavigate();
   const { globalPreferences } = useContext(GlobalPreferencesContext);
-
   const [errorMessage, setErrorMessage] = useState<string>();
   const auth0 = useAuth0();
+  // state is populated in NotFoundView
+  // state.origin holds a route that was initially requested,
+  // but was not found in the routes
+  const { state } = useLocation();
 
   useEffect(() => {
     const bases = globalPreferences.availableBases;
-    if (bases != null) {
+    if (bases && Array.isArray(bases)) {
       if (bases.length > 0) {
         const newBaseId = bases[0].id;
-        navigate(`/bases/${newBaseId}`);
+        // try to prepend /bases/:baseId to state.origin
+        navigate(
+          state && state.origin ? `/bases/${newBaseId}${state.origin}` : `/bases/${newBaseId}`,
```

Comment:
Can I test this somehow?

Reply:
just request any route like /qrreader or /test in the Browser and it should prepend /bases/1

Reply:
thanks, works :+1: